x-pack/plugins/apm/server/routes/alerts/rule_types/error_count/get_error_count_chart_preview.ts

```diff
@@ -38,24 +39,34 @@ export async function getTransactionErrorCountChartPreview({
     start,
     end,
     groupBy: groupByFields,
+    useFilterQuery,
+    filterQuery,
   } = alertParams;
 
   const allGroupByFields = getAllGroupByFields(
     ApmRuleType.ErrorCount,
     groupByFields
   );
 
+  const termFilterQuery =
+    !useFilterQuery || useFilterQuery === 'false'
+      ? [
+          ...termQuery(SERVICE_NAME, serviceName, {
+            queryEmptyString: false,
+          }),
+          ...termQuery(ERROR_GROUP_ID, errorGroupingKey, {
+            queryEmptyString: false,
+          }),
+          ...environmentQuery(environment),
+        ]
+      : [];
```

Comment:
We already have [`kqlFilter`](https://github.com/elastic/kibana/blob/933d97ff6d6c90976e947e088708df4127954a87/x-pack/plugins/observability/server/utils/queries.ts#L63) helper to parse a kql string.

Also, we don't need `useFilterQuery`, just `filterQuery` (which should be renamed to `kqlFilter`)

```suggestion
  const termOrKqlFilter =
    !kqlFilter
      ? [
          ...termQuery(SERVICE_NAME, serviceName, {
            queryEmptyString: false,
          }),
          ...termQuery(ERROR_GROUP_ID, errorGroupingKey, {
            queryEmptyString: false,
          }),
          ...environmentQuery(environment),
        ]
      : kqlQuery(kqlFilter);
```